crescent/commands/errors.py

```diff
@@ -1,27 +1,23 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Callable, Generic, Protocol, Type, TypeVar, overload
+from typing import TYPE_CHECKING, Any, Awaitable, Callable, Type, cast, overload
 
 from crescent.internal.meta_struct import MetaStruct
 
 if TYPE_CHECKING:
     from crescent.context import Context
+    from crescent.typedefs import ERROR, ErrorHandlerCallbackT
 
 
-ERROR = TypeVar("ERROR", bound=Exception, contravariant=True)
-
-
-class ErrorHandlerProto(Protocol, Generic[ERROR]):
-    async def __call__(self, /, *, exc: ERROR, ctx: Context) -> None:
-        ...
+_ErrorHandlerCallback = Callable[["ERROR", "Context"], Awaitable[None]]
 
 
 @overload
 def catch(
     exception: Type[ERROR], /
 ) -> Callable[
-    [ErrorHandlerProto[ERROR] | MetaStruct[ErrorHandlerProto[ERROR], Any]],
-    MetaStruct[ErrorHandlerProto[ERROR], Any],
+    [ErrorHandlerCallbackT | MetaStruct[_ErrorHandlerCallback, Any]],
+    MetaStruct[_ErrorHandlerCallback, Any],
```

Comment:
Is there a reason you removed the `ERROR` generic arg? Also why do you use ErrorHandlerCallbackT in some places and then _ErrorHandlerCallback in others?

Reply:
Using facing functions and internal functions need a different type. The other issue you had is the reason.